packages/design-system/src/components/Table/Table.jsx

```diff
@@ -0,0 +1,151 @@
+import React, { useEffect, useRef, useState } from 'react';
+import Alert from '../Alert/Alert';
+import PropTypes from 'prop-types';
+import TableCaption from './TableCaption';
+import classNames from 'classnames';
+
+function debounce(fn, ms) {
+  let timer;
+  return () => {
+    clearTimeout(timer);
+    timer = setTimeout(() => {
+      timer = null;
+      fn.apply(this, arguments);
+    }, ms);
+  };
+}
+
+export const Table = ({
+  className,
+  responsiveTable,
+  striped,
+  scrollable,
+  scrollableActiveAlert,
+  children,
+  ...attributeOptions
+}) => {
+  const container = useRef(null);
+  // The captionID is calculated and stored as init value of a ref.
+  // This ensures that the ID remains constant for all renders.
+  const captionID = useRef('caption-' + Math.random().toString(36).substr(2, 9));
+  const [tabIndex, setTabIndex] = useState(null);
+
+  // Use the Effect hook to listen for window resize event.
+  // React execute useEffect() on each re-render and returns a function that cleans up the event listener from the render.
+  // https://reactjs.org/docs/hooks-effect.html
+  useEffect(() => {
+    // Set `tabIndex` to `0` if the table width is wider than the viewport and Table is set for horizontal scrolling.
+    if (scrollable) {
+      const { scrollWidth, clientWidth } = container.current;
+      const scrollableActive = scrollWidth > clientWidth;
+      setTabIndex(scrollableActive ? '0' : null);
+
+      const debouncedHandleResize = debounce(function handleResize() {
+        const { scrollWidth, clientWidth } = container.current;
+        const scrollableActive = scrollWidth > clientWidth;
+        setTabIndex(scrollableActive ? '0' : null);
+      }, 500);
+
+      // Create window resize event listener that calls the debouncedHandleResize function
+      window.addEventListener('resize', debouncedHandleResize);
+
+      // Specify how to clean up after this effect to avoid memory leak by removing establishd event listeners
+      return () => {
+        window.removeEventListener('resize', debouncedHandleResize);
+      };
+    }
+  });
+
+  const classes = classNames(
+    'ds-c-table',
+    striped ? 'ds-c-table--striped' : null,
+    responsiveTable ? `ds-c-table--stacked-${responsiveTable}` : null,
+    className
+  );
+
+  // `tabIndex` state is needed to make table container focusable and to display scrollable message when table width exceeds viewport.
+  // Set attribute `tabIndex = 0` to make table container focusable to enable keyboard support of using the arrow keys.
+  // Also, provide context for screen reader users as they are able to focus on the region.
+  // Do this by using table's <caption> to label the scrollable region using aira-labelleby
+  const attributescrollable = scrollable && {
+    className: 'ds-c-table__wrapper',
+    role: 'group',
+    'aria-labelledby': captionID.current,
+    'aria-live': 'polite',
```

Comment:
In talking to @1Copenut, this is needed to let screen readers know that the area is scrollable, because they may not be able to see the page change. So it's good to have it, but it doesn't work because the alert is outside of the table. So it doesn't pick up the addition.
- We could move the alert in the table caption. May not be ideal in all instances of the component because not all tables need a caption. But maybe if there wasn't one, this was added in there?
- Or we could put the `aria-live` on the wrapper div and put the notice just inside that `div`, before the table. That may be cleaner code wise.

Reply:
Added alert to caption.

Reply:
I like the idea of putting the alert in the caption dynamically.

Reply:
I would also consider making `scrollableCaption` required, this way `scrollable` tables will always have a caption. It already has a default value and users can always override it.